aptos-move/framework/supra-framework/sources/poel/poel.move

```diff
@@ -0,0 +1,308 @@
+    use supra_framework::table;
+    use supra_framework::smart_table::SmartTable;
+    //Errors:
+
+    /// thrown when the allowed signer is not the caller (owner)
+    const ERR_NOT_OWNER: u64 = 0;
+
+    /// thrown when the wrong asset weight gets calculated
+    const ERR_WRONG_WEIGHT: u64 = 1;
+
+    /// thrown when the length of the supplied collaterixation weight argument vector does not match table entries
+    const ERR_WRONG_CWV_LENGTH: u64 = 2;
+
+    /// thrown when the length of the desirability score argument vector does not match table entries
+    const ERR_WRONG_DESIRABILITY_SCORE_LEN: u64 = 3;
+
+    /// thrown when the asset id supiled is not present in the TotalLiquidityTable
+    const ERR_ASSET_NOT_PRESENT: u64 = 4;
+
+    /// thrown If coefficient_X < coefficient_b
+    const ERR_COEFA_LT_COEFB: u64 = 5;
+ 
+    // Structs:
+ 
+     /// This struct contains contains the field attributes for each asset in the liquidity table
+     struct LiquidityTableItems {
+        ///Asset_name
+        asset_name: vector<u8>,
+        ///asset_supply: Total amount of the asset that has been bridged to create the iAsset.
+        asset_supply: u64,
+        ///iAsset_supply: Total supply of iAssets generated through the scheme.
+        iAsset_supply: u64,
+        ///collaterisation_rate: Specific rate indicating the collateral security of the iAsset.
+        collateralisation_rate: u64,
+        ///desired_weights: Weights reflecting the asset's strategic importance.
+        desired_weight: u64,
+        ///desirability_score: Attractiveness score of the asset.
+        desirability_score: u64,
+        ///asset_price: Current market price of the asset.
+        asset_price: u64,
+    }
+ 
+    /// This struct aggregates comprehensive liquidity metrics and operational indices for assets managed within the system.
+    struct TotalLiquidity {
+        //TotalLiquidityTable: Maps each AssetID to a LiquidityTableItems struct
+        TotalLiquidityTable: SmartTable<u64, LiquidityTableItems>,
+        //Index of the last observable lockup cycle (OLC), crucial for the minting of preminted tokens and the withdrawal of unlocked tokens.
+        current_olc_index: u64,
+        //Represents the nominal value of all assets submitted to the system, indicating the total economic stake.
+        total_nominal_value: u64
+    }
+ 
+    /// This struct is applied for the dynamic adjustment of the system's operational parameters. 
+    /// It allows for flexible management of financial metrics, adapting to market conditions or strategic shifts in policy.
+    struct MutableParameters { 
+        coefficient_b: u64,
+        coefficient_X: u64,
+        coefficient_rho: u64, 
+        min_collateralisation: u64 
+    }
+
+
+    struct StakingPoolMap {
+        //Address of the staking pool.
+        pool_address: address,
+        //The amount of Supra delegated to the validator at this pool.
+        delagated_amount: u64,
```

Comment:
Not sure what this means, does it mean the total of `active`, and `pending_inactive`?